python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -890,6 +914,37 @@ def _fast_slow_function_call(func: Callable, /, *args, **kwargs) -> Any:
                 # try slow path
                 raise Exception()
             fast = True
+            if cudf_pandas_debugging:
```

Comment:
```suggestion
            if _env_get_bool("CUDF_PANDAS_DEBUGGING", False):
```

We can just centralize the changes to just `_fast_slow_function_call` instead of trying to read the env variable at all of it's callers.

Reply:
My understanding is we wanted it to be default argument in `fast_slow_function_call`. Its definitely cleaner without is though. I'm curious what @bdice since he originally wanted it to be an argument, but I think that was before I switched from an option --> an environment variable.

Reply:
We have been trying to keep the proxy machinery generic. If we move a check for `CUDF_PANDAS_DEBUGGING` into the `fast_slow_function_call`, then we lose the generic nature of the proxy (it's specific to cudf.pandas). I am generally in favor of centralizing this but it's a tradeoff with keeping the machinery generic. @galipremsagar I'll defer to whatever you prefer here.

Reply:
Upon looking at this PR again, I see that we don't have a great way to inject this debugging variable name. Let's centralize it as Prem suggested for now. Ideally all the specific name choices like `CUDF_PANDAS_DEBUGGING` would live in pandas-specific wrappers, like `cudf/pandas/_wrappers/pandas.py`, and wouldn't appear in `fast_slow_proxy.py` where the machinery is supposed to be generic.